mne/io/eeglab/eeglab.py

```diff
@@ -443,3 +518,54 @@ def __init__(self, input_fname, events=None, event_id=None, tmin=0,
             reject=reject, flat=flat, reject_tmin=reject_tmin,
             reject_tmax=reject_tmax, add_eeg_ref=False, verbose=verbose)
         logger.info('Ready.')
+
+
+def _read_eeglab_events(eeg, event_id=dict(), event_id_func=lambda t:
+                        int("".join([x for x in t if x.isdigit()]))):
+    """Create events array from EEGLAB structure by looking them up in the
+    event_id, trying to reduce them to their integer part otherwise, and
+    entirely dropping them (with a warning) if this is impossible.
+    Returns a 3x3 array of zeros if no events are found."""
+    types = [event.type for event in eeg.event]
+    latencies = [event.latency for event in eeg.event]
+
+    not_in_event_id = set([x for x in types if x not in event_id])
```

Comment:
hmm ... if the user gives an `event_id` dict, we should use it completely, no? Basically, you have one code path if there is an `event_id` provided and another code path if nothing is provided. Does that sound okay to you?